subcommands/version/updates.go

```diff
@@ -91,6 +99,72 @@ func (f *FioctlUpdateFinder) FindLatest() (*FioctlUpdate, error) {
 	return update, nil
 }
 
+type progressBar struct {
+	total   int64
+	written int64
+	sha     hash.Hash
+	buff    bytes.Buffer
+}
+
+func (p *progressBar) Write(buff []byte) (n int, err error) {
+	if _, err = p.sha.Write(buff); err != nil {
+		return 0, err
+	}
+	n, err = p.buff.Write(buff)
+	p.written += int64(n)
+	// Print 20 dashes total
+	dashes := 100 * p.written / p.total / 5
+	spaces := 20 - dashes
+	progress := strings.Repeat("*", int(dashes)) + strings.Repeat(" ", int(spaces))
+	fmt.Printf("[%s] %d%%\r", progress, dashes*5)
+	return
+}
+
+func (u FioctlUpdate) Do() error {
+	exe, err := os.Executable()
```

Comment:
"Executable returns the path name for the executable that started the current process. There is no guarantee that the path is still pointing to the correct executable. If a symlink was used to start the process, depending on the operating system, the result might be the symlink or the path it pointed to. If a stable result is needed, path/filepath.EvalSymlinks might help."

Reply:
it works correctly. my home setup has /usr/local/bin/fioctl as a symlink to /home/doanac/code/fioctl.....

Reply:
ok.